### PR TITLE
BACKLOG-14186 add breadcrumb in content editor

### PR DESCRIPTION
**Pull request by Taybou**
## JIRA

https://jira.jahia.org/browse/BACKLOG-14186

## Tests

The following are included in this PR
- [ ] Unit Tests (Most changes _should_ have unit tests)
- [ ] Integration Tests

## Checklist

I have considered the following implications of my change: 

- [ ] Security (in particular for changes to authentication, authorization, data fetching, ...)
- [ ] Performance
- [ ] Migration
- [ ] Code maintainability

## Documentation

- [ ] Inline documentation
- [ ] Internal Documentation (wiki)
- [ ] User-facing Documentation

### CODE REVIEW

**Review by dgriffon — COMMENTED** [review]
Globally it seems odd for jContent to expose the breadcrumb:
- There is no usage of that code in jContent, making it difficult to understand. 
- it adds complexity to the component that would have been avoid if the CE logic had been put in its own implementation. 

If we choose to keep that implementation:
- jContent should also use the registered component. 
- the component should be more generic and take a path as parameter

**Review by tdraier — CHANGES_REQUESTED** [review]
You cannot use the regsitry to expose a component. Share the different parts in external libraries or recopy the code.
If you really need to share the logic, create a hook in data-helper (as we did for preview).

File: src/javascript/JContent/ContentRoute/ContentBreadcrumb/ContentPath/ContentPath.container.jsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import React, {useMemo} from 'react';
+import PropTypes from 'prop-types';
 import {useDispatch, useSelector} from 'react-redux';
 import {useQuery} from '@apollo/react-hooks';
 
 import {cmGoto} from '~/JContent/JContent.redux';
 import {GetContentPath} from './ContentPath.gql-queries';
 import ContentPath from './ContentPath';
+import JContentConstants from '../../../JContent.constants';
```
Comment (cedricmoitrier) [inline src/javascript/JContent/ContentRoute/ContentBreadcrumb/ContentPath/ContentPath.container.jsx:9]:
```suggestion
import JContentConstants from '~/JContent/JContent.constants';
```

File: src/javascript/JContent/ContentRoute/ContentBreadcrumb/ContentPath/ContentPath.container.jsx [hunk 2]
```diff
@@ -47,20 +49,34 @@ const ContentPathContainer = () => {
...
+        }));
     };
 
     if (error) {
         console.log(error);
     }
 
     const node = data?.jcr?.node || {};
-    const items = useMemo(() => getItems(node), [node]);
+    const items = useMemo(() => getItems(externalPath, node), [node, externalPath]);
```
Comment (cedricmoitrier) [inline src/javascript/JContent/ContentRoute/ContentBreadcrumb/ContentPath/ContentPath.container.jsx:70]: You should use the node returned by useQuery here... I don't understand the need for passing externalPath.

**Review by cedricmoitrier — CHANGES_REQUESTED** [review]

**Comment by Taybou** [comment]
Thank you for the review, I'll close this PR.

**Pull request closed without merging**